internal/controller/dpuoperatorconfig_controller.go

```diff
@@ -36,8 +36,39 @@ import (
 // DpuOperatorConfigReconciler reconciles a DpuOperatorConfig object
 type DpuOperatorConfigReconciler struct {
 	client.Client
-	Scheme      *runtime.Scheme
-	BinDataPath string
+	Scheme          *runtime.Scheme
+	BinDataPath     string
+	imagePullPolicy string
+}
+
+type DpuOperatorConfigReconcilerBuilder struct {
```

Comment:
What do we gain with this new infrastructure?  We now have 2 objects with the same variables...

Reply:
Good point. I see how the main intent of this code got lost and how this is actually worst. I will remove it.